packages/flutter_driver/test/flutter_driver_test.dart

```diff
@@ -35,8 +35,8 @@ void main() {
       when(mockVM.isolates).thenReturn([mockIsolate]);
       when(mockIsolate.loadRunnable()).thenReturn(mockIsolate);
       when(mockIsolate.invokeExtension(any, any))
-          .thenReturn(new Future.value({'status': 'ok'}));
-      vmServiceConnectFunction = (_) => new Future.value(mockClient);
+          .thenReturn(new Future<Map<String, dynamic>>.value(<String, String>{'status': 'ok'}));
```

Comment:
`flutter_driver` actually does that, which is what this test is testing. The VM service gives us a JSON blob and then that blob is deserialized into structured data. The mock for the VM service, however, has to deal with JSON blobs.

I'm not strongly against adding types. Special cases means more complexity in tooling. I'm guessing the linter is much simpler if it has to enforce the same rule in all code.